### PR TITLE
WIP Numerical

### PR DESCRIPTION
**Pull request by susuhahnml**

### CODE REVIEW

File: asp_interactive_configuration/examples/metro/constraints.lp [hunk 1]
```diff
@@ -3,85 +3,78 @@
...
+ooasp_cv(only_one_wagon,new_object,"Only one wagon allowed, found also wagon {}",(W2,)):-
+    ooasp_isa(wagon,new_object),
     ooasp_isa(wagon,W2),
-    W1!=W2.
+    new_object>W2.
 
-% standingroom > 0 -> count(handrail)>0
-ooasp_cv(standingroom_iff_handrail,W,"Standing room > 0 iff there is a handrail v1 {} {} {} {}",(new_object,W,STANDINGROOM,COUNT)):-
+% Handrail required if standing_room>0
+ooasp_cv(standingroom_iff_handrail,W,"Handrail required if stanfing rum is {}",(STANDINGROOM,new_object)):-
```
Comment (rtaupe) [inline asp_interactive_configuration/examples/metro/constraints.lp:26]:
```suggestion
ooasp_cv(standingroom_iff_handrail,W,"Handrail required if standing room is {}",(STANDINGROOM,new_object)):-
```

File: asp_interactive_configuration/examples/metro/fconstraints.lp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+% TODO: still missing is the length_mm attribute
+%
+#program domain(new_object, cls).
+
+ooasp_potential_cv(nrofseats_eq_attribute_nrseats).
+ooasp_potential_cv(standingroom_iff_handrail).
+
+% Paggensers number is calculated
+ooasp_cv(nrpassengers_neq_nrseats_plus_standingroom,new_object,"Number of passenges not adding up",(new_object,)):-
```
Comment (rtaupe) [inline asp_interactive_configuration/examples/metro/fconstraints.lp]:
```suggestion
ooasp_cv(nrpassengers_neq_nrseats_plus_standingroom,new_object,"Number of passengers not adding up",(new_object,)):-
```

File: asp_interactive_configuration/ooasp/config.py [hunk 3]
```diff
@@ -394,6 +396,12 @@ def remove_value(self, object_id:id, attr_name:str)->List[Predicate]:
         q = self.fb.query(AttributeValue)
         q = q.where(((AttributeValue.object_id == object_id) & (AttributeValue.attr_name == attr_name)))
         values = list(q.all())
+        
+        # print(f"object_id :{object_id}")
+        # print(f"attr_name :{attr_name}")
+        # print(list(self.fb.query(AttributeValue).where(AttributeValue.object_id == object_id).all()))
+        # print("To remove")
+        # print(values)
```
Comment (rtaupe) [inline asp_interactive_configuration/ooasp/config.py:443]: please remove commented-out code

File: asp_interactive_configuration/ooasp/interactive.py [hunk 4]
```diff
@@ -154,6 +172,16 @@ def __str__(self):
 
         return s
 
+    def theory_on_model(self, model):
+        model = model.model_
+        # self.theory.on_model(model)
```
Comment (rtaupe) [inline asp_interactive_configuration/ooasp/interactive.py]: please do not include commented-out code

File: asp_interactive_configuration/ooasp/interactive.py [hunk 5]
```diff
@@ -473,16 +499,20 @@ def _brave_config_as_options(self) -> dict:
...
         options = {}
-        for i in range(1,config.domain_size+1):
+        for i in range(1,self.config.domain_size+1):
             options[i]=[]
 
-        options = self._add_objects_to_dict(config, options)
-        options = self._add_attributes_to_dict(config, options)
-        options = self._add_associations_to_dict(config, options)
+        if self.brave_config is None:
+            # config = self.config
```
Comment (rtaupe) [inline asp_interactive_configuration/ooasp/interactive.py]: please do not include commented-out code

**Review by rtaupe — CHANGES_REQUESTED** [review]
I reviewed only part of the PR. Is it already ready for merge after review? If no, please convert it to a draft.

**Comment by susuhahnml** [comment]
The fixes to these comments were integrated into the cautious branch.

**Pull request closed without merging**